### PR TITLE
fix(docs): update it-threat-detection.ipynb to SDK v8

### PR DESCRIPTION
**Pull request by jhamon**
## Summary

- Updates the IT threat detection notebook to use Pinecone Python SDK v8
- Modernizes API key handling with getpass fallback pattern
- Uses `pc.has_index()` instead of deprecated `list_indexes().names()` pattern
- Removes outdated TensorFlow/Keras version pins
- Updates stale documentation link and removes time-sensitive language

## Related

- Resolves [SDK-134](https://linear.app/pinecone-io/issue/SDK-134/update-it-threat-detectionipynb-to-sdk-v8)

## Test plan

- [ ] Verify notebook JSON is valid
- [ ] Confirm Colab badge links work correctly
- [ ] Check that code cells use correct SDK v8 patterns

---

> [!NOTE]
> **Medium Risk**
> Mostly documentation/notebook updates, but adds a new `scripts/ticketbot.py` automation CLI that spawns subprocess worker pools, manages git worktrees, and runs the `agent` command with retries—misconfiguration could impact local repos or CI runners.
> 
> **Overview**
> **Notebook update:** Refreshes `learn/security/it-threat-detection.ipynb` to Pinecone Python SDK v8 usage (e.g., `pip install pinecone`, `getpass` API key fallback, `pc.has_index()`), loosens dependency pins, and updates minor docs/formatting.
> 
> **New automation tooling:** Adds `scripts/ticketbot.py`, a `click`-based CLI that orchestrates multiple worker pools (`tb-pick-work`, `tb-iterate-prs` with modulo sharding, `tb-merge-ready`) by creating per-worker `git worktree`s, invoking `agent` with logging and exponential-backoff retries, and supporting graceful drain on SIGTERM/CTRL-C.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 6bcbb96f8c06fff2d3eca89d2f8e21fa010ec934. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: learn/security/it-threat-detection.ipynb [hunk 1]
```diff
@@ -1,1822 +1,1793 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "r3g-b61IywpQ"
+   },
+   "outputs": [],
+   "source": [
+    "!pip install -qU pip python-dateutil tensorflow scikit-learn matplotlib seaborn"
```
[the remark this answers is not in the document]
Comment (jhamon) [reply learn/security/it-threat-detection.ipynb]: Fixed in 9a25514 - updated the import to use `from tensorflow.keras.utils import normalize` since standalone keras was removed from dependencies.
